src/parser/jobs/nin/Kassatsu.js

```diff
@@ -0,0 +1,174 @@
+import {Trans, Plural} from '@lingui/react'
+import React, {Fragment} from 'react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+export default class Kassatsu extends Module {
+	static handle = 'kassatsu'
+	static dependencies = [
+		'combatants',
+		'suggestions',
+	]
+
+	_kassatsuSpent = false
+	_kassatsuWastes = 0
+	_kassatsuUses = {
+		[ACTIONS.FUMA_SHURIKEN.id]: 0,
+		[ACTIONS.KATON.id]: 0,
+		[ACTIONS.RAITON.id]: 0,
+		[ACTIONS.HYOTON.id]: 0,
+		[ACTIONS.HUTON.id]: 0,
+		[ACTIONS.DOTON.id]: 0,
+		[ACTIONS.SUITON.id]: 0,
+		[ACTIONS.RABBIT_MEDIUM.id]: 0,
+	}
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player', abilityId: Object.keys(this._kassatsuUses).map(Number)}, this._onNinjutsuCast)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.KASSATSU.id}, this._onRemoveKassatsu)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onNinjutsuCast(event) {
+		const abilityId = event.ability.guid
+		if (this.combatants.selected.hasStatus(STATUSES.KASSATSU.id)) {
+			this._kassatsuUses[abilityId]++
+			this._kassatsuSpent = true
+		}
+	}
+
+	_onRemoveKassatsu() {
+		if (!this._kassatsuSpent) {
+			this._kassatsuWastes++
+		}
+
+		// Reset the flag for the next time it's cast
+		this._kassatsuSpent = false
+	}
+
+	_onComplete() {
+		if (this._kassatsuWastes > 0) {
```

Comment:
Not gonna block on it, but this seems like a _lot_ of potential suggestion spam if someone's dicking up their kassatsus.

Is it feasible to consolidate them a bit, or would it make the suggestions themselves meaningless? Iuno enough about NIN to say much.

Reply:
I was hemming and hawing over keeping them grouped as a single suggestion, but I broke them out for two reasons - varying severity and ease of translation. Like, using Kassatsu on Suiton is way less bad than using it on Huton (especially if setting up a TA so they don't fall off schedule). I could maybe consolidate some of them, but in practice I wouldn't expect more than one, maybe two of these suggestions to show per log, and if more than two _do_ show, the player really needs to get their shit together.